### PR TITLE
Setting up infrastructure for the detector description

### PR DESCRIPTION
**Pull request by jlingema**
With this PR we (@clementhelsens and myself) tried to come up with a schema that allows to have several detector-baselines for the different accelerators. At the same time we would like to allow switching out sub-detectors to allow flexibility for studies. 

The reasoning and naming scheme for the folder-structure can be found in [this mark-down](https://github.com/jlingema/FCCSW/blob/d3633059f3f37fe1c9788729e77f17736a3e6521/Detector/doc/DD4hepInFCCSW.md#fccsw-folder-structure). The plan is to present this in an upcoming SW meeting to and discuss changes needed by detector experts here (hence the "[WIP]").

### CODE REVIEW

File: Detector/DetCommon/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+find_package(Geant4)
+include(${Geant4_USE_FILE})
+
+set(CMAKE_MODULE_PATH  ${CMAKE_MODULE_PATH}  ${DD4hep_ROOT}/cmake )
+include( DD4hep )
+
+find_package(ROOT COMPONENTS MathCore GenVector Geom REQUIRED)
+
+gaudi_add_module(DetCommon
+		  src/*.cpp
```
Comment (hegner) [inline Detector/DetCommon/CMakeLists.txt]: Don't mix tabs and spaces ;-)

File: Detector/DetFCChhECalSimple/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+find_package(Geant4)
+include(${Geant4_USE_FILE})
+
+set(CMAKE_MODULE_PATH  ${CMAKE_MODULE_PATH}  ${DD4hep_ROOT}/cmake )
+include( DD4hep )
+
+find_package(ROOT COMPONENTS MathCore GenVector Geom REQUIRED)
+
+gaudi_add_module(DetFCChhECalSimple
+		  src/*.cpp
```
Comment (hegner) [inline Detector/DetFCChhECalSimple/CMakeLists.txt]: again - tabs and spaces

File: Detector/doc/DD4hepInFCCSW.md [hunk 3]
```diff
@@ -357,4 +365,32 @@ export LIBGL_ALWAYS_INDIRECT=1
...
+The detector descriptions and geometry services / tools can be found in the `Detector` package. The FCCSW components that provide the geometry to the remaining packages can be found in:
+- DetInterface (virtual interface classes)
+- DetComponents (actual implementations of the services / tools)
+
+Extensions of the DD4hep description are found in:
+- DetExtensions (for reconstruction)
+- DetSensitive (custom sensitive volumes)
+
+### The master XMLs
+The descriptions are placed in sub-folders of the Detector package. We plan to have the possibility of having several baseline detectors for the different accelerators. For each such baseline we will have a sub-directory `"DetFCC" + Flavour + "Baseline" + X` where X will be an increasing index and `Flavour = ee, eh, hh`, selected to describe the accelerator the baseline is meant for (for example `DetFCChhBaseline1`). In each baseline description we expect to have a `compact` directory with:
```
Comment (hegner) [inline Detector/doc/DD4hepInFCCSW.md]: Can you add a little ascii graphic of the directory layout? Should be easier to understand then.

Comment (jlingema) [inline Detector/doc/DD4hepInFCCSW.md]: Added one a bit further up as an overview.

**Comment by hegner** [comment]
Thanks for the updates!

**Pull request merged**